internal/commands/config/view_profile_internal.go

```diff
@@ -23,35 +22,46 @@ func RunInternalConfigViewProfile(args []string) (err error) {
 	}
 
 	// Validate the profile name
-	err = profiles.GetMainConfig().ValidateExistingProfileName(pName)
+	err = profiles.GetKoanfConfig().ValidateExistingProfileName(pName)
 	if err != nil {
 		return fmt.Errorf("failed to view profile: %w", err)
 	}
 
-	msgStr := fmt.Sprintf("Configuration for profile '%s':\n", pName)
+	// Set the active profile to the one specified
+	if err := profiles.GetKoanfConfig().ChangeActiveProfile(pName); err != nil {
```

Comment:
Why setting an active profile change in view profile command?

Reply:
This sets the profile if the user runs `pingcli config view-profile <profile name>`, it's never written to the config file for this command.

Reply:
I think the next function to read a specific profile name makes this unnecessary.

Reply:
line 48 is calling the `KoanfValueFromOption` method which requires the "main" `KoanfConfig` value